snuba/migrations/runner.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+from datetime import datetime
+from enum import Enum
+
+from snuba.clusters.cluster import ClickhouseClientSettings, get_cluster, CLUSTERS
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations.groups import get_group_loader, MigrationGroup
+
+logger = logging.getLogger("snuba.migrations")
+
+LOCAL_TABLE_NAME = "migrations_local"
+DIST_TABLE_NAME = "migrations_dist"
+
+# Currently only local migrations are supported
+TABLE_NAME = LOCAL_TABLE_NAME
+
+
+class Status(Enum):
+    NOT_STARTED = "not_started"
+    IN_PROGRESS = "in_progress"
+    COMPLETED = "completed"
+
+
+class Runner:
+    def run_migration(self, group: MigrationGroup, migration_id: str) -> None:
+        """
+        Run a single migration given its ID and marks the migration as complete.
+        """
+        assert all(
+            cluster.is_single_node() for cluster in CLUSTERS
+        ), "Cannot run migrations for multi node clusters"
+
+        logger.info(f"Running migration: {group} {migration_id}")
+
+        migration = get_group_loader(group).load_migration(migration_id)
```

Comment:
Shouldn't you mark the migration as "in progress" before you run the first operation ?

Reply:
We previously discussed just doing this for data migrations but I think it might be worth doing in other cases as well. It is definitely not going to work in this case since the migration creates the table that would need to be written to

Reply:
So what is the plan ?

Reply:
I can think of a few different options, such as skipping it for the system migrations and keeping it for everything else. I'd rather come back to this decision later, unless you feel we need it for this PR?

Reply:
please add a TODO and it will be ok.

Reply:
Sure, done